File: src/main/java/com/beat/domain/member/application/dto/MemberLoginResponse.java
package com.beat.domain.member.application.dto;

public record MemberLoginResponse(
	String accessToken,
	String nickname,
	String role
) {
	public static MemberLoginResponse of(
		final String accessToken,
		final String nickname,
		final String role
	) {
		return new MemberLoginResponse(accessToken, nickname, role);
	}
}

File: apis/src/main/kotlin/com/beat/apis/member/api/response/MemberLoginResponse.kt
package com.beat.apis.member.api.response

import com.beat.apis.member.application.result.LoginSuccessResult

@ConsistentCopyVisibility
data class MemberLoginResponse private constructor(
    val accessToken: String?,
    val nickname: String?,
    val role: String?,
) {
    companion object {
        fun from(result: LoginSuccessResult): MemberLoginResponse = MemberLoginResponse(
            accessToken = result.accessToken,
            nickname = result.nickname,
            role = result.role,
        )
    }
}
